### PR TITLE
Matrix Chain Multiplication in C++

### PR DESCRIPTION
**Pull request by Meghag5**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes

### Description

(Write your answer here.)
Given a sequence of matrices, find the most efficient way to multiple these matrices together. We have to find the way which costs minimum.
Input:
 Size of array -
 4
 Input Array -
 arr = {10, 30, 5, 60}
Output:
Minimum costs -
4500

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#6016

### CODE REVIEW

**Comment by Meghag5** [comment]
@HarshCasper pls review

File: C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp [hunk 1]
```diff
@@ -1,101 +1,69 @@
-// Matrix chain multiplication algorithm 
-#include <bits/stdc++.h> 
-using namespace std; 
+/*
+Question:- Given a sequence of matrices, find the most efficient way to multiple these matrices together. We have to find the way which costs minimum.
+How are we defining coat of matrix multiplication ?
+=> Let's asssume two matrices are there of order 10X30 and 30X20 so cost of multiplication of them will be 10*30*20 , we will take the common order element only once .
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp]: Remove this and write
 - Add Multi-Line Comments 
- Add a beginner-friendly description at the top

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp (hunk 1 above)
Comment (Meghag5) [reply C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp]: @ankitaggarwal23 changes done pls review

File: C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp [hunk 2]
```diff
@@ -1,101 +1,70 @@
...
+Space Complexity: O(N2)
+Input:
+ Size of array -
+4
+ Input Array -
+arr = {10, 30, 5, 60}
+Output:
+Minimum costs -
+4500
+*/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp]: Rewrite this as,

```
/*

Input: 
Size of array -                                                                                                                                 
4                                                                                                                                               
Input Array -                                                                                                                                   
10 30 5 60  

Output:                                                                                                                                    
Minimum costs -                                                                                                                                 
4500

Time Complexity: O(N ^ 3)
Space Complexity: O(N ^ 2)
*/
```

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp (hunk 2 above)
Comment (Meghag5) [reply C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp]: done

Comment (Meghag5) [reply C-Plus-Plus/dp/Matrix_Chain_Multiplication.cpp]: pls have a look

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Why the problem is in dp section? It is actually a mathematics problem though you used dp kindly shift in math section also readme link produces produces error. Check properly

**Comment by Meghag5** [comment]
updated pls check @ankitaggarwal23

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Why updating matrix chain twice in readme?

**Comment by Meghag5** [comment]
> Why updating matrix chain twice in readme?

what I am not able to understand?

**Comment by HarshCasper** [comment]
> > Why updating matrix chain twice in readme?
> 
> what I am not able to understand?

Get on this [page](https://github.com/TesseractCoding/NeoAlgo/blob/f196a2930ec36342c0ecc1725aeb9f73cfd93db3/C-Plus-Plus/README.md) and you will find two references for the Matrix Chain Multiplication. One of the link is dead so resolve that up.

**Comment by Meghag5** [comment]
pls check now @HarshCasper @ankitaggarwal23 changes done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**